SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayClient.java

```diff
@@ -0,0 +1,235 @@
+                    if (status == SPAY_READY) {
+                        getBraintreeSupportedSamsungPayCards(new GetAcceptedCardBrandsCallback() {
+                            @Override
+                            public void onResult(@Nullable List<SpaySdk.Brand> acceptedCardBrands, @Nullable Exception error) {
+                                if (acceptedCardBrands != null) {
+                                    boolean isReadyToPay = !acceptedCardBrands.isEmpty();
+
+                                    if (isReadyToPay) {
+                                        braintreeClient.sendAnalyticsEvent("samsung-pay.is-ready-to-pay.ready");
+                                        callback.onResult(true, null);
+                                    } else {
+                                        braintreeClient.sendAnalyticsEvent("samsung-pay.request-card-info.no-supported-cards-in-wallet");
+                                        Exception noCardsInWalletError =
+                                            new SamsungPayException(SamsungPayError.SAMSUNG_PAY_NO_SUPPORTED_CARDS_IN_WALLET);
+                                        callback.onResult(false, noCardsInWalletError);
+                                    }
+                                } else {
+                                    callback.onResult(false, error);
+                                }
+                            }
+                        });
+                    } else {
+                        switch (status) {
+                            case SPAY_NOT_READY:
+                                braintreeClient.sendAnalyticsEvent("samsung-pay.is-ready-to-pay.not-ready");
+                                break;
+                            case SPAY_NOT_SUPPORTED:
+                                braintreeClient.sendAnalyticsEvent("samsung-pay.is-ready-to-pay.device-not-supported");
+                                break;
+                        }
+
+                        // still callback error here; it may contain additional information about why Samsung Pay is not enabled
+                        callback.onResult(false, samsungPayError);
+                    }
+                } else {
+                    callback.onResult(false, samsungPayError);
+                }
+            }
+        });
+    }
+
+    private void getSamsungPayStatus(final GetSamsungPayStatusCallback callback) {
+        getInternalClient(new GetSamsungPayInternalClientCallback() {
+            @Override
+            public void onResult(@Nullable SamsungPayInternalClient internalClient, @Nullable Exception error) {
+                if (internalClient != null) {
+                    internalClient.getSamsungPayStatus(callback);
+                } else {
+                    callback.onResult(null, error);
+                }
+            }
+        });
+    }
+
+    private void getBraintreeSupportedSamsungPayCards(final GetAcceptedCardBrandsCallback callback) {
```

Comment:
Can we make this method name align with the internal client method and callback name?

Reply:
Most def.

Reply:
Done!